src/index.ts

```diff
@@ -1,10 +1,11 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import { createIoCContainer } from './ioc';
+import type { Users } from './services/users';
+import type { User } from './types';
 
-import type { User, ApiConfig } from './types';
+const ioc = createIoCContainer();
+ioc.register('config', (window as any).__CONFIG__.api);
 
-const renderUsers = async (config: ApiConfig) => {
-  const usersService = new Users(config);
+const renderUsers = async (usersService: Users) => {
```

Comment:
you don't need to pass usersService as an argument. you can resolve it inside the renderUsers function.